archive/100-theorems-list/37_solution_of_cubic.lean

```diff
@@ -0,0 +1,195 @@
+/-
+Copyright (c) 2022 Jeoff Lee. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jeoff Lee
+-/
+import tactic.basic
+import ring_theory.roots_of_unity
+import ring_theory.polynomial.cyclotomic.basic
+
+/-!
+# The Solution of a Cubic
+
+This file proves Theorem 37 from the [100 Theorems List](https://www.cs.ru.nl/~freek/100/).
+
+In this file, we give the solutions to the cubic equation `a * x^3 + b * x^2 + c * x + d = 0` 
+over a field `K` that has characteristic neither 2 nor 3, that has a third primitive root of
+unity, and in which certain other quantities admit square and cube roots.
+
+This is based on the [Cardano's Formula](https://en.wikipedia.org/wiki/Cubic_equation#Cardano's_formula).
+
+## Main statements
+
+- `cubic_eq_zero_iff` : gives the roots of the cubic equation
+- ```
```

Comment:
```suggestion
```
It looks like there was some errant formatting in the `suggestion` block. I'll just fix it if you don't mind